scipy/stats/_continuous_distns.py

```diff
@@ -3991,7 +4022,7 @@ def _check_fit_input_parameters(dist, data, args, kwds):
     floc = kwds.get('floc', None)
     fscale = kwds.get('fscale', None)
 
-    num_shapes = len(dist.shapes) if dist.shapes else 0
+    num_shapes = len(dist.shapes.split(",")) if dist.shapes else 0
```

Comment:
nit: should the `len(args) > num_shapes:` check be moved adjacent to this?

Reply:
I like the idea of having all of the pieces of code that raise errors next to each other, but am open to moving it if you feel strongly.